### ISSUE TITLE
Synthesized audio has noise

### ISSUE THREAD

**Issue by MuyangDu** [fatchord/WaveRNN#196]
Thanks for the great work. I have trained wavernn on Databaker mandarin dataset(about 12 hours) for about 500K steps in MOL mode. But the synthesized audio has a lot of pitter patter noise. The only change I made in hparams is set fmin to 95 and fmax to 7600 when computing mel spectrograms during preprocessing. Do you have any ideas why there is noise in the synthesized audio? Does this noise has anything to do with fmin and fmax? A synthesized sample is attached below.  Thanks @fatchord 
[synthesized.wav.zip](https://github.com/fatchord/WaveRNN/files/4779113/synthesized.wav.zip)

**Comment by sih4sing5hong5** [comment]
How about trying RAW mode? More robust https://github.com/fatchord/WaveRNN/issues/159#issuecomment-561091073

**Comment by MuyangDu** [comment]
> How about trying RAW mode? More robust [#159 (comment)](https://github.com/fatchord/WaveRNN/issues/159#issuecomment-561091073)

Thanks @sih4sing5hong5 , Yes, I will try RAW mode once the MOL mode reaches 1000K steps.  Currently I have trained for about 800K step in MOL mode and the noise is still there.

**Comment by mindmapper15** [comment]
Usually, MOL sounds better than RAW mode when these models converged perfectly because of the quantization error on RAW mode.

What about disabling batched generation mode? Did you try that?

**Comment by MuyangDu** [comment]
> Usually, MOL sounds better than RAW mode when these models converged perfectly because of the quantization error on RAW mode.
> 
> What about disabling batched generation mode? Did you try that?

Yes. I have tried disabing batch mode in MOL mode but the noise is still there. And I have also tried the RAW mode and the noise is gone and the synthesized audio in RAW mode is as good as the original audio. I guess it is because I don't have enough data for MOL mode to converge to a good point.

**Comment by sih4sing5hong5** [comment]
> Usually, MOL sounds better than RAW mode when these models converged perfectly because of the quantization error on RAW mode.
> 
> What about disabling batched generation mode? Did you try that?

Is there any reference about MOL training? How to coverage or how much data does training need?
Thanks.

**Comment by mindmapper15** [comment]
> > Usually, MOL sounds better than RAW mode when these models converged perfectly because of the quantization error on RAW mode.
> > What about disabling batched generation mode? Did you try that?
> 
> Is there any reference about MOL training? How to coverage or how much data does training need?
> Thanks.

In my case, I've trained WaveRNN MOL with LJSpeech Dataset, which is the dataset of 24 hours of single female english speaker speech.

Hyperparameter I used:

```
# Settings for all models
sample_rate = 16000
n_fft = 1024
fft_bins = n_fft // 2 + 1
num_mels = 80
hop_length = 200                    # 12.5ms - in line with Tacotron 2 paper
win_length = 800                   # 50ms - same reason as above
fmin = 0
min_level_db = -100
ref_level_db = 20
bits = 9                            # bit depth of signal
mu_law = True                       # Recommended to suppress noise if using raw bits in hp.voc_mode below
peak_norm = True                   # Normalize to the peak of each wav file

# WAVERNN / VOCODER ------------------------------------------------------------------------------------------------#

# Model Hparams
voc_mode = 'MOL'                    # either 'RAW' (softmax on raw bits) or 'MOL' (sample from mixture of logistics)
voc_upsample_factors = (5, 5, 8)    # NB - this needs to correctly factorise hop_length
voc_rnn_dims = 512
voc_fc_dims = 512
voc_compute_dims = 128
voc_res_out_dims = 128
voc_res_blocks = 10

# Training
voc_batch_size = 64
voc_lr = 1e-4
voc_pad = 2                         # this will pad the input so that the resnet can 'see' wider than input length
voc_seq_len = hop_length * 40      # must be a multiple of hop_length
voc_clip_grad_norm = 4              # set to None if no gradient clipping needed

```

With this setting, around 500 epoch (approx 98k step), the quality of generated sound pretty decent.
And this is the generated sample from tacotron output with 1500 epoch of WaveRNN MOL model (approx 294k step)

[LJSpeech Inference.zip](https://github.com/fatchord/WaveRNN/files/4850699/LJSpeech.Inference.zip)

I used voc_seq_len = 8000 for better sound quality but I think 4000 sample is enough.
I checked validation loss of both 4000 and 8000 sample model but there was no significant difference.

And I trained WaveRNN model with tacotron output (args.gta) not the mel spectrogram of actual audio speech data.

**Comment by sih4sing5hong5** [comment]
Thank you for quick reply.

Our hyperparameters:
```

# DSP --------------------------------------------------------------------------------------------------------------#

# Settings for all models
sample_rate = 16000
n_fft = 2048
fft_bins = n_fft // 2 + 1
num_mels = 80
hop_length = 200                    # 12.5ms - in line with Tacotron 2 paper
win_length = 800                    # 50ms - same reason as above
fmin = 40
min_level_db = -100
ref_level_db = 20
bits = 9                            # bit depth of signal
mu_law = True                       # Recommended to suppress noise if using raw bits in hp.voc_mode below
peak_norm = False                   # Normalise to the peak of each wav file


# WAVERNN / VOCODER ------------------------------------------------------------------------------------------------#


# Model Hparams
voc_mode = 'RAW'                    # either 'RAW' (softmax on raw bits) or 'MOL' (sample from mixture of logistics)
voc_upsample_factors = (4, 5, 10)   # NB - this needs to correctly factorise hop_length
voc_rnn_dims = 512
voc_fc_dims = 512
voc_compute_dims = 128
voc_res_out_dims = 128
voc_res_blocks = 10

# Training
voc_batch_size = 32
voc_lr = 1e-4
voc_checkpoint_every = 25_000
voc_gen_at_checkpoint = 5           # number of samples to generate at each checkpoint
voc_total_steps = 1_000_000         # Total number of training steps
voc_test_samples = 50               # How many unseen samples to put aside for testing
voc_pad = 2                         # this will pad the input so that the resnet can 'see' wider than input length
voc_seq_len = hop_length * 5        # must be a multiple of hop_length
voc_clip_grad_norm = 4              # set to None if no gradient clipping needed
```

Most of them are default of this repository.
The `fmin`, `voc_mode`,  `voc_batch_size`, `voc_upsample_factors` and `voc_seq_len` are different from yours.
And the length of our corpus is only 4~5 hours. 
We will try new parameters.

Thanks again.

**Comment by theAayushbajaj** [comment]
@sih4sing5hong5 I also tried the RAW mode after MOL mode didn't converge pertaining to less data (~6hrs). RAW mode is still at 250k but the target wav files also have some noise in them. Did it happen to you as well?

**Comment by sih4sing5hong5** [comment]
Our corpus contains sentences only. We trained successfully. If diverge, just retrain again.

**Comment by theAayushbajaj** [comment]
> > Usually, MOL sounds better than RAW mode when these models converged perfectly because of the quantization error on RAW mode.
> > What about disabling batched generation mode? Did you try that?
> 
> Yes. I have tried disabing batch mode in MOL mode but the noise is still there. And I have also tried the RAW mode and the noise is gone and the synthesized audio in RAW mode is as good as the original audio. I guess it is because the training data is not enough (about 12 hours) for the MOL mode to converge to a good point.

@MuyangDu facing the same issue, how did you solve this?
